### ISSUE TITLE
Add ss.beta() class

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#630]
Handle parsing of networks etc. automatically to avoid the messiness of `validate_beta()` and add an extra layer of checking.

**Comment by cliffckerr** [comment]
Replace this:
```py
    hiv = ss.HIV(beta={'mf': [ss.beta(0.0008), ss.beta(0.0004)]}, init_prev=0.2) # TODO: beta should wrap the other way
    syph = ss.Syphilis(beta={'mf': [ss.beta(0.1), ss.beta(0.05)]}, init_prev=0.05)
```
with
```py
    hiv = ss.HIV(beta=ss.beta(mf=[0.0008, 0.0004]), init_prev=0.2) # TODO: beta should wrap the other way
    syph = ss.Syphilis(beta=ss.beta(mf=[0.1, 0.05]), init_prev=0.05)
```

**Comment by cliffckerr** [comment]
Closed by #623
